### PR TITLE
fix: request clone error

### PR DESCRIPTION
**Pull request by brenelz**
Closes https://github.com/denoland/deno/issues/15858
 
I am a bit unsure about the abortSignal logic

### CODE REVIEW

**Review by littledivy — COMMENTED** [review]
Thanks, please add a test case for https://github.com/denoland/deno/issues/15858 in `cli/tests/unit/flash_test.ts`

**Comment by brenelz** [comment]
Great idea... a test case has been added

File: ext/fetch/23_request.js [hunk 1]
```diff
@@ -161,7 +161,7 @@
   function cloneInnerRequest(request) {
     const headerList = [
       ...new SafeArrayIterator(
-        ArrayPrototypeMap(request.headerList, (x) => [x[0], x[1]]),
+        ArrayPrototypeMap(request.headerList || [], (x) => [x[0], x[1]]),
```
Comment (bartlomieju) [inline ext/fetch/23_request.js:164]: This seems off... In what situation would `request.headerList` be undefined?

Comment (brenelz) [reply ext/fetch/23_request.js:164]: thats a good quesiton. Maybe i'm treating the symptom and not the root cause

Comment (brenelz) [reply ext/fetch/23_request.js:164]: not sure if this is correct but adding ` headerList: headersCb()`to `fromFlashRequest` might be the real fix?

**Pull request closed without merging**